packages/block-editor/src/components/url-input/index.js

```diff
@@ -18,233 +18,286 @@ import {
 	withSpokenMessages,
 	Popover,
 } from '@wordpress/components';
-import { withInstanceId, withSafeTimeout, compose } from '@wordpress/compose';
-import { withSelect } from '@wordpress/data';
+import { useInstanceId, withSafeTimeout, compose } from '@wordpress/compose';
+import { useSelect } from '@wordpress/data';
 import { isURL } from '@wordpress/url';
 
-class URLInput extends Component {
-	constructor( props ) {
-		super( props );
-
-		this.onChange = this.onChange.bind( this );
-		this.onFocus = this.onFocus.bind( this );
-		this.onKeyDown = this.onKeyDown.bind( this );
-		this.selectLink = this.selectLink.bind( this );
-		this.handleOnClick = this.handleOnClick.bind( this );
-		this.bindSuggestionNode = this.bindSuggestionNode.bind( this );
-		this.autocompleteRef = props.autocompleteRef || createRef();
-		this.inputRef = createRef();
-		this.updateSuggestions = debounce(
-			this.updateSuggestions.bind( this ),
-			200
-		);
+const REQUEST_DEBOUNCE_DELAY = 200;
 
-		this.suggestionNodes = [];
+function defaultRenderControl( { controlProps, inputProps, isLoading } ) {
+	return (
+		<BaseControl { ...controlProps }>
+			<input { ...inputProps } />
+			{ isLoading && <Spinner /> }
+		</BaseControl>
+	);
+}
 
-		this.isUpdatingSuggestions = false;
+function defaultRenderSuggestions( {
+	suggestions,
+	suggestionsListProps,
+	handleSuggestionClick,
+	buildSuggestionItemProps,
+	className,
+	selectedSuggestion,
+} ) {
+	return (
+		<Popover position="bottom" noArrow focusOnMount={ false }>
+			<div
+				{ ...suggestionsListProps }
+				className={ classnames(
+					'block-editor-url-input__suggestions',
+					`${ className }__suggestions`
+				) }
+			>
+				{ suggestions.map( ( suggestion, index ) => (
+					<Button
+						{ ...buildSuggestionItemProps( suggestion, index ) }
+						key={ suggestion.id }
+						className={ classnames(
+							'block-editor-url-input__suggestion',
+							{
+								'is-selected': index === selectedSuggestion,
+							}
+						) }
+						onClick={ () => handleSuggestionClick( suggestion ) }
+					>
+						{ suggestion.title }
+					</Button>
+				) ) }
+			</div>
+		</Popover>
+	);
+}
 
-		this.state = {
-			suggestions: [],
-			showSuggestions: false,
-			selectedSuggestion: null,
+function URLInput( {
+	value,
+	onChange,
+	label,
+	className,
+	isFullWidth,
+	placeholder = __( 'Paste URL or type to search' ),
+	disableSuggestions,
+	speak,
+	debouncedSpeak,
+	__experimentalRenderControl: renderControl = defaultRenderControl,
+	__experimentalRenderSuggestions: renderSuggestions = defaultRenderSuggestions,
+	__experimentalFetchLinkSuggestions,
+	__experimentalHandleURLSuggestions: handleURLSuggestions = false,
+	__experimentalShowInitialSuggestions: showInitialSuggestions = false,
+	setTimeout,
+	parentAutocompleteRef,
```

Comment:
I believe you wanted to get `autocompleteRef` and rename to `parentAutocompleteRef`, correct?